### DISCUSSION TITLE
Is there a possibility to request mentions like requesting the AuthorFeed

### DISCUSSION THREAD

**Discussion by martinentner** [bluesky-social/atproto#1842] [How to do it?] [answered]
Hello,

next to requesting the AuthorFeed, i'd also like to request all the posts containing a mention of "my" account.

The only way i found so far is the notifications endpoint but this endpoint
- also delivers likes, follows etc. and especially when it comes to likes it can be thousands which leads to a lot of unnecessary requests just to find all the latest mentions
- does not deliver `embed` on top-level, just the `embed` within the `record` which contains way fewer data. In some cases there are workarounds to still get the most important data about the attachment in other cases - especially when it comes to embedded posts - it requires another request to fetch the data because only the "ID"s are provided.

In case there is no dedicated endpoint it would already help to have a filter possibility for the notifications endpoint against the `reason` and in the very best case, additionally the top-level `embed` as part of the response.

Best

Martin

**Comment by bnewbold** [comment, marked as answer]
Great question!

This is not a functionality in the Bluesky API (`bsky.app` Lexicons) right now.

Filtering the notifications based on `reason` has been requested by a couple folks and we'll probably add something there.

cc: @pfrazee for thoughts/status on filtering notifications

I think for your specific use-case the best option would be extending the post search API to allow a filter query on posts which mention a specific account. That would work for any account (which may have some moderation and harassment considerations to take in to account), and would likely handle complexities like handle/DID changes.

The whole team is pretty focused on large tasks related to federation and scaling, and I would guess a new functionality like this is unlikely to get shipped until that is complete, unfortunately. I think you would need to iterate through the full notification feed, and then re-fetch any relevant mention records separately, for now.

**Comment by treeder** [comment]
Is there a way to do this with `searchPosts` yet or is `listNotifications` still the only way?  I tried an empty `q` parameter with `mentions` parameter, but it doesn't work.
